virtualcluster/pkg/syncer/resources/pod/mutatorplugin/podrootcacertmutator.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutatorplugin
+
+import (
+	"sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/constants"
+	"sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/conversion"
+	"sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/syncer/util/featuregate"
+	uplugin "sigs.k8s.io/cluster-api-provider-nested/virtualcluster/pkg/util/plugin"
+)
+
+func init() {
+	MutatorRegister.Register(&uplugin.Registration{
+		ID: "00_PodRootCACertMutator",
+		InitFn: func(ctx *uplugin.InitContext) (interface{}, error) {
+			return &PodRootCACertMutatorPlugin{}, nil
+		},
+	})
+}
+
+type PodRootCACertMutatorPlugin struct{}
+
+// This mutator will automatically reassign configmap references for configmaps named
+// kube-root-ca.crt in the pod spec, these places are
+// * volumes
+// * env
+// * envFrom
+func (pl *PodRootCACertMutatorPlugin) Mutator() conversion.PodMutator {
+	return func(p *conversion.PodMutateCtx) error {
+		if featuregate.DefaultFeatureGate.Enabled(featuregate.RootCACertConfigMapSupport) {
```

Comment:
let's flip the if and do fast return if the feature is not enabled? You could even probably add this check before the plugin registration

Reply:
Great idea returning early. 
I'm a little worried about doing that in the init phase incase the feature gates are loaded after, since init is on import… am I wrong about that?

Reply:
I don't think we can cause we import them from cmd/vc-syncer and in main() from there we also start to load the flags,  I think fast exiting when featuregate is false is enough.